File: src/main/java/Solution_1207_UniqueNumberOfOccurrences.java
import java.util.*;

public class Solution_1207_UniqueNumberOfOccurrences {
    public static void main(String[] args) {
        var arr = new int[]{1, 2, 2, 1, 1, 3, 2};
        System.out.println(uniqueOccurrences(arr));
    }

    private static boolean uniqueOccurrences(int[] arr) {
        var map = new HashMap<Integer, Integer>();

        Arrays.stream(arr).forEach(num -> {
            map.compute(num, (key, value) -> value == null ? 0 : value + 1);
        });

        var unique = new HashSet<>(map.values());

        return unique.size() == map.size();
    }
}

File: src/main/kotlin/Solution_1207_UniqueNumberOfOccurrences.kt
/**
 * 1207 Unique Number of Occurrences
 *
 * Easy
 *
 * Given an array of integers arr, return true if the number of occurrences of each value in the array is unique or false otherwise.
 *
 * Example 1:
 *
 * Input: arr = [1,2,2,1,1,3]
 * Output: true
 * Explanation: The value 1 has 3 occurrences, 2 has 2 and 3 has 1. No two values have the same number of occurrences.
 *
 * Example 2:
 *
 * Input: arr = [1,2]
 * Output: false
 *
 * Example 3:
 *
 * Input: arr = [-3,0,1,-3,1,1,1,-3,10,0]
 * Output: true
 */
fun main() {
    val arr = intArrayOf(1, 2, 2, 1, 1, 3, 2)
    println(uniqueOccurrences(arr))
}

fun uniqueOccurrences(arr: IntArray): Boolean {
    val map = mutableMapOf<Int, Int>()

    // добавляем в Map каждое число из массива и подсчитываем их количество
    // num - ключ, количество вхождений - значение
    arr.forEach { num ->
        map.compute(num) { _, count -> (count ?: 0) + 1 }
    }

    // преобразовываем в Set, чтобы оставит только уникальные значения
    val unique = map.values.toSet()

    // сравниваем количество значений в Map и в unique
    return map.size == unique.size
}